internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -1200,3 +1202,211 @@ func (s *Server) Memory(ctx context.Context, in *empty.Empty) (reply *machine.Me
 
 	return reply, err
 }
+
+// EtcdMemberList implements the machine.MachineServer interface.
+func (s *Server) EtcdMemberList(ctx context.Context, in *machine.EtcdMemberListRequest) (reply *machine.EtcdMemberListResponse, err error) {
+	client, err := etcd.NewClientFromControlPlaneIPs(ctx, s.Controller.Runtime().Config().Cluster().CA(), s.Controller.Runtime().Config().Cluster().Endpoint())
+	if err != nil {
+		return nil, err
+	}
+
+	// nolint: errcheck
+	defer client.Close()
+
+	resp, err := client.MemberList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var members []string
+
+	for _, member := range resp.Members {
+		members = append(members, member.GetName())
+	}
+
+	reply = &machine.EtcdMemberListResponse{
+		Messages: []*machine.EtcdMemberList{
+			{
+				Members: members,
+			},
+		},
+	}
+
+	return reply, nil
+}
+
+// EtcdMemberRemove implements the machine.MachineServer interface.
+//
+// nolint: gocyclo
+func (s *Server) EtcdMemberRemove(ctx context.Context, in *machine.EtcdMemberRemoveRequest) (reply *machine.EtcdMemberRemoveResponse, err error) {
```

Comment:
nit, but should it be `EtcdMemberRemoveSelf`, as it only supports removing this node from the members?

Reply:
How about `EtcdForfeitLeadership`?

Reply:
I think this is about removing a member, but not really leadership?

I'm not a wordsmit, could be `EtcdLeave` ?

Reply:
Fixed.